### DISCUSSION TITLE
Unable to read variable set by set-env in github action

### DISCUSSION THREAD

**Discussion by gautamkrishnar** [community/community#26093] [Actions]
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/gautamkrishnar/blog-post-workflow/actions/runs/205462576" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/8397274/181095188-45eb7a24-818f-4a54-9d15-763aba12aeca.png" class="thumbnail onebox-avatar" width="398" height="398">

<h3><a href="https://github.com/gautamkrishnar/blog-post-workflow/actions/runs/205462576" target="_blank" rel="nofollow noopener">gautamkrishnar/blog-post-workflow</a></h3>

<p>Show your latest blog posts from any sources or StackOverflow activity on your GitHub profile/project readme automatically using the RSS feed - gautamkrishnar/blog-post-workflow</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Yaml:</p>
<pre><code class="lang-auto">name: Release to GitHub
on:
  workflow_dispatch:
  workflow_run:
    workflows: ["Version bump"]
    branches: [master]
    types:
      - completed
jobs:
  version-bump:
    name: Release to github releases
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Set env
        run: echo ::set-env name=pkg_version::$(awk '/version/{gsub(/("|",)/,"",$2);print $2}' package.json)
      - name: Create Release
        id: create_release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: ${{ pkg_version }}
          release_name: ${{ pkg_version }}
          body: |
            Changes in this Release
            - Change this
          draft: false
          prerelease: false
</code></pre>

**Comment by gautamkrishnar** [comment]
<p>I had to use this to fix the eror:</p>
<pre><code class="lang-auto">name: Publish release to GitHub
on:
  workflow_dispatch:
  workflow_run:
    workflows: ["Version bump"]
    branches: [master]
    types:
      - completed
jobs:
  version-bump:
    name: Release to github releases
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Set env
        run: echo ::set-output name=pkg_version::$(awk '/version/{gsub(/("|",)/,"",$2);print $2}' package.json)
        id: env_out
      - name: Create Release
        id: create_release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: ${{ steps.env_out.outputs.pkg_version }}
          release_name: ${{ steps.env_out.outputs.pkg_version }}
          body: |
            Changes in this Release
            - Change this
          draft: false
          prerelease: false
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="gautamkrishnar" data-post="1" data-topic="127140">
<div class="title">
<div class="quote-controls"></div>
 gautamkrishnar:</div>
<blockquote>
<pre><code class="lang-auto">        with:
          tag_name: ${{ pkg_version }}
          release_name: ${{ pkg_version }}
</code></pre>
</blockquote>
</aside>
<p>You need to reference the <code>env</code> context to read environment variables outside the shell:</p>
<pre><code class="lang-yaml">        with:
          tag_name: ${{ env.pkg_version }}
          release_name: ${{ env.pkg_version }}
</code></pre>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/gautamkrishnar">@gautamkrishnar</a>,</p>
<p>After you use <code>set-env</code>, the variable has been added to env context. You can add below code for confrimation.</p>
<pre><code class="lang-auto">      - name: Dump env context
        env:
          ENV_CONTEXT: ${{ toJson(env) }}
        run: echo "$ENV_CONTEXT"
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181095192-9bc57bb2-733f-4bb4-8c74-1edf86c5b7c8.png" alt="image" data-base62-sha1="1lMpEcVtt4xtyoXTzuWUoG8EAPs" width="552" height="118"></p>
<p>Hence, you can use ${{ env.pkg_version }} to invoke it.</p>
<p>Thanks</p>

**Comment by gautamkrishnar** [comment]
<p>Thanks, it is now solved.</p>
